frontend/src/components/EditableSelect.tsx

```diff
@@ -0,0 +1,151 @@
+import classNames from 'classnames';
+import {
+  FC,
+  useState,
+  ComponentPropsWithoutRef,
+  Dispatch,
+  SetStateAction,
+  useEffect,
+} from 'react';
+import Select from 'react-select';
+import { CloseButton, ConfirmButton, EditButton } from './forms/SvgButton';
+import { LoadingSpinner } from './LoadingSpinner';
+import { FieldError } from './forms/FormField';
+import css from './EditableSelect.module.scss';
+
+const classes = classNames.bind(css);
+
+interface Option {
+  value: number;
+  label: string;
+}
+
+interface WithButtonsProps {
+  onOk: (newValue: number, fieldId: string) => Promise<string | void>;
+  value: Option;
+  fieldId: string;
+  format: string | undefined;
+  options: Option[];
+}
+
+const withButtons = (Component: typeof EditableSelect) => ({
+  onOk,
+  value,
+  fieldId,
+  format,
+  options,
+}: WithButtonsProps) => {
+  const [editOpen, setEditOpen] = useState(false);
+  const [editValue, setEditValue] = useState<Option | undefined>(undefined);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleConfirm = () => {
+    setIsLoading(true);
+    onOk(editValue!.value, fieldId).then((returnedError) => {
+      if (returnedError) setErrorMessage(returnedError);
+      else setEditOpen(false);
+      setIsLoading(false);
+    });
+  };
+
+  const handleCancel = () => {
+    setErrorMessage('');
+    setEditValue(undefined);
+    setEditOpen(false);
+  };
+
+  return (
+    <div
+      className={classes(css.withButtonsContainer, {
+        [css.withButtonsContainerEditMode]: editOpen,
+      })}
+    >
+      <Component
+        id={fieldId}
+        editOpen={editOpen}
+        editValue={editValue}
+        onChange={(newValue) => {
+          setEditValue(newValue as Option);
+          setErrorMessage('');
+        }}
+        error={{ message: errorMessage, setMessage: setErrorMessage }}
+        isLoading={isLoading}
+        value={value}
+        format={format}
+        options={options}
+      />
+      {editOpen ? (
+        <div className={classes(css.buttonsDiv)}>
+          <CloseButton onClick={handleCancel} />
+          {errorMessage === '' && <ConfirmButton onClick={handleConfirm} />}
```

Comment:
```suggestion
          <ConfirmButton onClick={handleConfirm} />
```
This isn't doing any form validation (each option is valid), so only errors should be from submitting the choice.
If any validation, it could be checked here that `editValue !== undefined`.